[CI] revisit shellcheck excludes

[Currently, the Gluon CI excludes](https://github.com/freifunk-gluon/gluon/blob/e2de86ce9dfaaf3f5aac890bd1a0cedbacf25dc2/scripts/lint-sh.sh#L20)

- [SC2039](https://www.shellcheck.net/wiki/SC2039): `This warning has been retired in favor of individual SC3xxx warnings for each individual issue. Removed in V0.7.2 - 2021-04-19`
- [SC1091](https://www.shellcheck.net/wiki/SC1091): Needs a revisit
- [SC2155](https://www.shellcheck.net/wiki/SC2155): Needs a revisit
- [SC2034](https://www.shellcheck.net/wiki/SC2034): Needs a revisit:
  Probably required for scripts that are embedded within other scripts. Maybe this can be handled differently to find undefined variables?
- [SC3043](https://www.shellcheck.net/wiki/SC3043): Can be deprecated with shellcheck v0.10.0 with busybox sh support
- [SC3037](https://www.shellcheck.net/wiki/SC3037): Can be deprecated with shellcheck v0.10.0 with busybox sh support
- [SC3057](https://www.shellcheck.net/wiki/SC3057): Can be deprecated with shellcheck v0.10.0 with busybox sh support

cc @AiyionPrime 

EDIT by @rotanid 
we also need to make sure, we apply the same rules for other repositories like packages and community-packages as well.

--- a/package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh
+++ b/package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh
@@ -2,25 +2,18 @@
 
 # shellcheck disable=SC1091
 
-# FIXME: The following lints should not need to be disabled
-# shellcheck disable=SC2034,SC2155
-
-PROTO_DEBUG=1
-
 . /lib/functions.sh
 . ../netifd-proto.sh
 init_proto "$@"
 
-WG=/usr/bin/wg
-
 proto_gluon_wireguard_init_config() {
-	proto_config_add_int index
-	proto_config_add_int mtu
+	:
 }
 
 interface_linklocal_from_wg_public_key() {
 	# We generate a predictable v6 address
-	local macaddr="$(printf "%s" "$1"|md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
+	local macaddr
+	macaddr="$(printf "%s" "$1"|md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
 	local oldIFS="$IFS"; IFS=':';
 	# shellcheck disable=SC2086
 	set -- $macaddr; IFS="$oldIFS"
@@ -31,14 +24,13 @@ proto_gluon_wireguard_setup() {
 	local config="$1"
 	local ifname="$2"
 
-	local index mtu
-	json_get_vars index mtu
-
-	local public_key="$(/lib/gluon/mesh-vpn/wireguard_pubkey.sh)"
+	local public_key
+	public_key="$(/lib/gluon/mesh-vpn/wireguard_pubkey.sh)"
 
 	# The wireguard proto itself can not be moved here, as the proto does not
 	# allow add_dynamic.
 
+	local wireguard_ip
 	wireguard_ip=$(interface_linklocal_from_wg_public_key "$public_key")
 
 	## Add IP

